pkgs/development/compilers/zig/default.nix

```diff
@@ -21,12 +21,27 @@ llvmPackages.stdenv.mkDerivation rec {
   ];
 
   preBuild = ''
-    export HOME=$TMPDIR;
+    export HOME=$TMPDIR
   '';
 
   checkPhase = ''
     runHook preCheck
-    ./zig test $src/test/stage1/behavior.zig
+    ./zig build -Dskip-non-native \
```

Comment:
why not `./zig build test -Dskip-non-native` ? It runs all the tests below.

also you can speed things up considerably if you are willing to skip release tests: `-Dskip-release`

Reply:
This was still a WIP PR; I split the tests up line-by-line because I was planning to disable them to track down which one was failing, and I needed to transfer them from one computer to another. I'm not sure on the general NixOS policy on skipping release tests, hence I left them in. IMO it's probably worth running them because binary caches exist and it feels safer.